lib/vagrant/action/builtin/disk.rb

```diff
@@ -0,0 +1,33 @@
+module Vagrant
+  module Action
+    module Builtin
+      class Disk
+        def initialize(app, env)
+          @app    = app
+          @logger = Log4r::Logger.new("vagrant::action::builtin::disk")
+        end
+
+        def call(env)
+          machine = env[:machine]
+          defined_disks = get_disks(machine, env)
+
+          # Call into providers machine implementation for disk management
+          #
+          # machine.provider.configure_disks(defined_disks)

```

Comment:
I'm imagining it'll probably be something like this? Or maybe this is just a "base" call and providers will have to override the builtin disk action? :thinking: